src/modules/wlr/taskbar.cpp

```diff
@@ -460,38 +460,78 @@ bool Task::operator!=(const Task &o) const
 
 void Task::update()
 {
-    if (!format_before_.empty()) {
-        text_before_.set_label(
-                fmt::format(format_before_,
-                    fmt::arg("title", title_),
-                    fmt::arg("app_id", app_id_),
-                    fmt::arg("state", state_string()),
-                    fmt::arg("short_state", state_string(true))
-                )
-        );
-        text_before_.show();
-    }
-    if (!format_after_.empty()) {
-        text_after_.set_label(
-                fmt::format(format_after_,
-                    fmt::arg("title", title_),
-                    fmt::arg("app_id", app_id_),
-                    fmt::arg("state", state_string()),
-                    fmt::arg("short_state", state_string(true))
-                )
-        );
-        text_after_.show();
-    }
+    bool markup = config_["markup"].isBool() ? config_["markup"].asBool() : false;
+    if (markup) {
```

Comment:
Rather than this big if/else, could you save the fmt::format and then doing an if/else with just the set_markup/set_label?

Reply:
Since the arguments to the format (title, app_id) also need to change, I did it this way to avoid making a copy of those strings unless they needed escaping. I could switch to always making local strings, using them for the format, and then choosing set_{markup,text} if you think that's clearer.